pulseaudio-control.bash

```diff
@@ -59,13 +61,44 @@ function getSinkName() {
 # "Sink #<index>" is used.
 function getNickname() {
     getSinkName "$1"
+    unset nickname
+
     if [ -n "${SINK_NICKNAMES[$sinkName]}" ]; then
         nickname="${SINK_NICKNAMES[$sinkName]}"
-    else
+    elif [ -n "$SINK_NICKNAME_PROP" ]; then
+        nickname="$(getNicknameFromProp "$SINK_NICKNAME_PROP" "$sinkName")"
+        # Cache that result for next time
+        SINK_NICKNAMES["$sinkName"]="$nickname"
+    fi
+
+    if [ -z "$nickname" ]; then
         nickname="Sink #$1"
     fi
 }
 
+# Gets sink nickname based on a given property.
+function getNicknameFromProp() {
+    local nickname_prop="$1"
+    local for_name="$2"
+
+    mapfile -t lines < <(pacmd list-sinks)
```

Comment:
one more thing: I think this would be better by redirecting stdin into the loop instead of creating a map:

```sh
while read -r file; do

done < <(pacmd list-sinks)
```

Reply:
Much better, thanks!